bcgov/HOUS-permit-portal
# Update step code compliance check logic

## Description

Adds support for multiple step code requirements when checking a permit application's step code for compliance. Refactors the compliance check so it checks all requirements and combines energy and zero carbon results together to facilitate multiple requirements.

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

[BPHH-1490](https://hous-bssb.atlassian.net/browse/BPHH-1490)

## Steps to QA

1. Login as a review manager and configure the step code requirements for the jurisdictions permit types
2. Login as a submitter and create an application for the jurisdiction. Add the energy step code to the application using the compliance tool.

Verify: 
- If the jurisdiction permit type is configured with more than one step requirement, the submitter should have the option to toggle between the step requirements on the energy step code tool inside the permit application form.
- if the jurisdiction permit type is configured with a single step requirement, the submitter should not see the radio group and just see the result for the single requirement. 

3. As the submitter, submit the permit application
4. As the review manager, open the submitted application and download the step code PDF

Verify:
- Step code PDF is generated correctly (e.g. compliance sections are populated) and displays the results as per the requirement selected when the permit application checklist was last saved (e.g. if a passing requirement was selected, it shows passing, if a failing requirement was selected, it shows failing). 

Uploading Screen Recording 2024-06-28 at 2.34.41 PM.mov…


## UI Accessibility Checklist



_Note the following checklist is not an exhaustive list. Check
the [WebAIM checklist](https://webaim.org/standards/wcag/checklist) for
a user-friendly checklist with more details. Aim to meet at least *AA* conformance level where applicable. Check
the [WCAG guidelines](https://webaim.org/standards/wcag/checklist) for the full guidelines._

      with [Wave Browser Extension](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh)
      for accessibility errors?

## General Checklist


## [optional] Are there any post-deployment tasks we need to perform?

n/a
